tests/components/unifi/test_button.py

```diff
@@ -75,3 +75,89 @@ async def test_restart_device_button(
     # Controller reconnects
     await websocket_mock.reconnect()
     assert hass.states.get("button.switch_restart").state != STATE_UNAVAILABLE
+
+
+async def test_power_cycle_poe(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, websocket_mock
+) -> None:
+    """Test restarting device button."""
+    config_entry = await setup_unifi_integration(
+        hass,
+        aioclient_mock,
+        devices_response=[
+            {
+                "board_rev": 3,
+                "device_id": "mock-id",
+                "ip": "10.0.0.1",
+                "last_seen": 1562600145,
+                "mac": "00:00:00:00:01:01",
+                "model": "US16P150",
+                "name": "switch",
+                "state": 1,
+                "type": "usw",
+                "version": "4.0.42.10433",
+                "port_table": [
+                    {
+                        "media": "GE",
+                        "name": "Port 1",
+                        "port_idx": 1,
+                        "poe_caps": 7,
+                        "poe_class": "Class 4",
+                        "poe_enable": True,
+                        "poe_mode": "auto",
+                        "poe_power": "2.56",
+                        "poe_voltage": "53.40",
+                        "portconf_id": "1a1",
+                        "port_poe": True,
+                        "up": True,
+                    },
+                ],
+            }
+        ],
+    )
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
```

Comment:
We should not need to access the controller since we know the host and site from the config entry data. I think we've talked about this before. :smile:

Reply:
Probably 😅 , thanks for your attention to details Martin! I will add this to my list of improvements :)

Reply:
https://github.com/home-assistant/core/pull/112275